jax/_src/numpy/lax_numpy.py

```diff
@@ -4406,10 +4421,11 @@ def _linspace(start: ArrayLike, stop: ArrayLike, num: int = 50,
   if issubdtype(dtype, integer) and not issubdtype(out.dtype, integer):
     out = lax.floor(out)
 
+  sharding = canonicalize_device_to_sharding(device)
   if retstep:
-    return lax.convert_element_type(out, dtype), delta
+    return lax_internal._convert_element_type(out, dtype, sharding=sharding), delta
```

Comment:
This is long enough now that it may be better to write it out only once, and then `return (result, delta) if retstep else result`